GameSync.Api/Middleware/ExceptionMiddlewareExtensions.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System.Net;
+using FluentValidation;
+using GameSync.Api.Shared.Middleware.Models;
+using GameSync.Domain.Shared.Exceptions;
+using Microsoft.AspNetCore.Diagnostics;
+
+namespace GameSync.Api.Shared.Middleware;
+
+/// <summary>
+/// Middleware for handling exceptions.
+/// </summary>
+public static class ExceptionMiddlewareExtensions
+{
+    /// <summary>
+    /// Configuration of exception middleware.
+    /// </summary>
+    /// <param name="app">Application builder.</param>
+    public static void ConfigureExceptionHandler(this IApplicationBuilder app)
+    {
+        app.UseExceptionHandler(appError =>
+        {
+            appError.Run(async context =>
+            {
+                context.Response.ContentType = "application/json";
+                var contextFeature = context.Features.Get<IExceptionHandlerFeature>();
+                if (contextFeature != null)
+                {
+                    string messagePrefix = $"Path: {context.Request.Path} IP: {context.Request.HttpContext.Connection.RemoteIpAddress} Message: "; // todo: body?
+                    string message = string.Empty;
+                    List<string> errorsList = new List<string>();
+                    switch (contextFeature.Error)
```

Comment:
Tutaj jeszcze będą dodane kody 401 i 403? Bo teraz jest obsłużony kod 400 oraz domyślnie 500 leci. Czy inne kody będziemy dodawać w miarę rozwoju aplikacji?

Reply:
Nie. Tutaj dodajemy obsługę wyjątków i tłumaczymy je na odpowiednie kody http. Jeżeli chcesz rzucić jakimś nietypowym kodem http np. z zakresu 4xx to tworzysz customowy wyjątek i go tutaj obsługujesz. Następnie rzucasz w handlerze